packages/widgets-internal/roi/aprHelper.ts

```diff
@@ -1,8 +1,9 @@
-import { Percent } from "@pancakeswap/sdk";
-import { formatFraction } from "@pancakeswap/utils/formatFractions";
-
-export function getAccrued(principal: number, apr: Percent, timesCompounded = 1, stakeFor = 1) {
-  const aprAsDecimal = parseFloat(formatFraction(apr.asFraction, 6) || "0");
+export function getAccrued(principal: number, apr: number, compoundEvery = 0, stakeFor = 1, compoundOn = false) {
   const daysAsDecimalOfYear = stakeFor / 365;
-  return principal * (1 + aprAsDecimal / timesCompounded) ** (timesCompounded * daysAsDecimalOfYear);
+  const timesCompounded = 365 / compoundEvery;
+  if (timesCompounded !== 0 && compoundEvery <= stakeFor && compoundOn) {
+    return principal * (1 + apr / timesCompounded) ** (timesCompounded * daysAsDecimalOfYear);
+  }
+
+  return principal + principal * apr * daysAsDecimalOfYear; // simple calc when not compounding
```

Comment:
Avoid floating numbers, pls use bigNumber to handle it.

Reply:
done